block-producer/src/batch_builder/batch.rs

```diff
@@ -113,6 +127,17 @@ impl TransactionBatch {
     pub fn created_notes_root(&self) -> Digest {
         self.created_notes_smt.root()
     }
+
+    fn calc_id(txs: &[SharedProvenTx]) -> BatchId {
```

Comment:
A couple nits:

1. Let's add a section separator above called "HELPER FUNCTIONS" as this is not a public accessor.
2. Let's call this function `compute_id()`.